### DISCUSSION TITLE
问几个问题

### DISCUSSION THREAD

**Discussion by RC1844** [xmake-io/xmake#4633] [Q&A] [answered]
1. 一个package附带了一些资源文件(图片之类的)，我拷贝到了包的bin目录(install时没有被拷贝)，如何在调试前将资源文件拷贝到targetdir？
2. 如何控制qt的qrc文件编译为rcc二进制文件？顺便，qt的ts翻译文件有相关支持吗？
3. 如何控制本地包的文件安装，我install时文件没有拷贝下面是包描述
```lua
    package("ZSVisual")
    on_fetch(function(package, opt)
        local path = os.scriptdir() .. "/ZSVisual"
        local linkdir = path .. "/Libs/Windows/x64"

        local result = {
            shared = true,
            links = { "DogLicense_X64", -- "ZsCAD_X64",
                "ZSCalib_X64", "ZSVisionCL_X64" },
            bindir = linkdir,
            linkdirs = linkdir,
            includedirs = path .. "/include"
        }
        package:addenv("PATH", result.bindir)
        return result
    end)

    on_test(function(package)
        assert(package:has_cxxtypes("ZSVisionCL", {
            includes = "ZSVisionCL.h"
        }))
    end)
    package_end()
```

**Comment by waruqi** [comment, marked as answer]
走 target:pkg() 访问包的安装目录来自己 copy。

```lua
add_requires("xxx")
target("test")
    after_install(function (target)
        local xxx = target:pkg("xxx")
        if xxx then
             local installdir = xxx:installdir()
             os.cp(...)
        end
    end)
    add_packages("xxx")
```

> 如何控制qt的qrc文件编译为rcc二进制文件？顺便，qt的ts翻译文件有相关支持吗？

qrc 目前不是原本就支持自动 rcc 编译么。。ts 的不清楚，应该还没支持

**Reply by RC1844** [reply]
> 走 target:pkg() 访问包的安装目录来自己 copy。

好的，了解。

> qrc 目前不是原本就支持自动 rcc 编译么。。ts 的不清楚，应该还没支持

qrc好像目前只能编译并合到程序中吧，rcc.exe还可以控制编译为一个单独的.rcc文件，qt可以动态加载卸载这个.rcc文件

**Reply by waruqi** [reply]
目前不支持，你得自己写 rule 处理。

**Reply by RC1844** [reply]
> 目前不支持，你得自己写 rule 处理。

我尝试在做原来的基础上增加binary处理，然后不知道怎么传参到buildcmd😂，我要指定输出路径，我尝试在add_files的时候传参，但是不知道怎么取出来🙃。还有就是qt.qrc规则好像有bug，如果有多个同名的qrc，前面的会被覆盖掉，因为里面用了basename

**Reply by waruqi** [reply]
set_values + target:values("")  传参，也可以 add_rules 里面传参后取，参考其他 rule 比如 protobuf rule

**Reply by waruqi** [reply]
bug 可以提 pr 过来修复

**Reply by RC1844** [reply]
> set_values + target:values("")  传参，也可以 add_rules 里面传参后取，参考其他 rule 比如 protobuf rule 

是用target:fileconfig这个吗？

**Reply by waruqi** [reply]
嗯

**Reply by RC1844** [reply]
我处理好了，需要提交上来吗？代码可能不太好。还是只修bug？

**Reply by waruqi** [reply]
只修 bug

**Reply by RC1844** [reply]
好的，pr提交了
